torchvision/transforms/functional_pil.py

```diff
@@ -30,7 +39,7 @@ def get_image_size(img: Any) -> List[int]:
 @torch.jit.unused
 def get_image_num_channels(img: Any) -> int:
     if _is_pil_image(img):
-        return 1 if img.mode == "L" else 3
+        return len(img.getbands())
```

Comment:
That's [a better way](https://stackoverflow.com/a/53880997) to get the number of channels. What we had earlier was wrong (for example when transparency is used).